### PR TITLE
New adapter: Appstock

### PR DESCRIPTION
**Pull request by imedvedko**
Docs: https://github.com/prebid/prebid.github.io/pull/4822

### CODE REVIEW

File: static/bidder-info/appstock.yaml
Comment (gargcreation1992) [inline static/bidder-info/appstock.yaml:3]: verification sent to this email address. Please reply with confirmed.

File: static/bidder-info/appstock.yaml
Comment (onkarvhanumante) [inline static/bidder-info/appstock.yaml:1]: verified endpoint is reachable

**Review by onkarvhanumante — DISMISSED** [review]
good to go after @gargcreation1992  approval

File: static/bidder-info/appstock.yaml
Comment (imedvedko) [reply static/bidder-info/appstock.yaml:3]: received

File: static/bidder-params/appstock.json
Comment (gargcreation1992) [inline static/bidder-params/appstock.json:13]: Could you please explain the usage of host information in bidder params?

Comment (imedvedko) [reply static/bidder-params/appstock.json:13]: There is information about subdomains and domains. We opted out of this decision at your request here: https://github.com/prebid/prebid-server/pull/2653/files#diff-267b5e4d799f4ba6dea77765c24de75d3a77e1c40f5936f9a0033260790ccb87R1
You can read more about it here: https://github.com/prebid/prebid-server/pull/2539#discussion_r1095070010

Comment (imedvedko) [reply static/bidder-params/appstock.json:13]: We also discussed it here: https://github.com/prebid/prebid-server/pull/2741#discussion_r1183673730
As noted in the docs PR https://github.com/prebid/prebid.github.io/pull/4297, we need to support the same parameters with our Prebid.js adapter. Our Prebid.js has Host parameter. For compatibility with our Prebid.js adapter parameters

Comment (gargcreation1992) [reply static/bidder-params/appstock.json:13]: Agreed. We discourage macros to be part of domain names based on the reasons of the comments that you have provided.
This is more of questioning regarding usage of host information as bidder param. Please try to explain the reasoning on  why would publisher require to enter `host` information given that usage is to do internal routing. This should be the concern of the bidder servers. I have linked the suggestions on how could you do the internal routing.

Comment (imedvedko) [reply static/bidder-params/appstock.json:13]: There are two reasons for this decision
1. `appstock` is just an `limelightDigital` adapter alias. `limelightDigital` is SaaS. We have a lot of clients. We need to know which client came to our servers. We use this parameter to determine which client came to our servers. We can create adapter for each of our clients. We were asked not to do that here: https://github.com/prebid/prebid-server/pull/2539#discussion_r1104911811
2. Backward compatibility https://github.com/prebid/prebid-server/pull/3054#discussion_r1309873081

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/4822

File: static/bidder-params/appstock.json
Comment (imedvedko) [reply static/bidder-params/appstock.json]: I removed host param from template and properties

**Review by onkarvhanumante — DISMISSED** [review]

File: static/bidder-params/appstock.json
Comment (gargcreation1992) [reply static/bidder-params/appstock.json]: @imedvedko thank you for being patient on this one.
The suggestion was to remove the host bidder param from the `limelightDigital` adapter and then from the aliases. It is inappropriate to remove the bidder param just from the alias since alias shares the same code as `limelightDigital`. 
Requesting you to please revert the changes in https://github.com/prebid/prebid-server/pull/3054/commits/76fabc4d81060e4dc78349cecc25a95ac08a1d74 commit. 
This is again a suggestion - Once `limelightDigital` bidder server is ready to internally route the traffic and not requiring the host param, you can come back and then change the bidder params from the parent as well as its aliases.

Comment (imedvedko) [reply static/bidder-params/appstock.json]: We can not remove `host` param from `limelightDigital` adapter because it is the client identificator. `limelightDigital` is SaaS. We have a lot of clients. We need to know which client came to our servers. We use `host` parameter to determine which client came to our servers. We don't use it as endpoint hostname, just url query parameter.
But I can remove `host` param from `appstock` alias because we already know that it is appstock by endpoint hostname

Comment (gargcreation1992) [reply static/bidder-params/appstock.json]: Since `appstock` is an alias of `limelightDigital`, we cannot have different bidder params. So, requesting you to revert the changes in https://github.com/prebid/prebid-server/commit/76fabc4d81060e4dc78349cecc25a95ac08a1d74 commit.

Comment (imedvedko) [reply static/bidder-params/appstock.json]: @gargcreation1992 Sorry for the flood of questions. Does you mean to revert https://github.com/prebid/prebid-server/pull/3054/commits/cc4e8f174ff15d4887b702e0e726774f6a8aa091 or https://github.com/prebid/prebid-server/pull/3054/commits/76fabc4d81060e4dc78349cecc25a95ac08a1d74 ? Could you double check please?

Comment (gargcreation1992) [reply static/bidder-params/appstock.json]: thanks for checking in.
Actually both. In https://github.com/prebid/prebid-server/commit/cc4e8f174ff15d4887b702e0e726774f6a8aa091 - the `host` param was removed. 
And in `https://github.com/prebid/prebid-server/commit/76fabc4d81060e4dc78349cecc25a95ac08a1d74` test cases were removed.

**Comment by gargcreation1992** [comment]
@imedvedko Do you also wanna revert https://github.com/prebid/prebid-server/pull/3054/commits/76fabc4d81060e4dc78349cecc25a95ac08a1d74 params test?

**Comment by imedvedko** [comment]
@gargcreation1992 test which I removed in https://github.com/prebid/prebid-server/commit/76fabc4d81060e4dc78349cecc25a95ac08a1d74 commit was added in https://github.com/prebid/prebid-server/commit/cc4e8f174ff15d4887b702e0e726774f6a8aa091 but I reverted https://github.com/prebid/prebid-server/commit/cc4e8f174ff15d4887b702e0e726774f6a8aa091 so I don't need to revert https://github.com/prebid/prebid-server/commit/76fabc4d81060e4dc78349cecc25a95ac08a1d74

**Comment by gargcreation1992** [comment]
> 61eabd4

Don't think you have completely reverted the changes in https://github.com/prebid/prebid-server/pull/3054/commits/61eabd47b04b38d40fb0bf8948a2d8dcd8f6a49a commit. Test cases are still missing. Could you please re-check? 
Also changes from all the commits do not show the test cases. https://github.com/prebid/prebid-server/pull/3054/files

**Comment by imedvedko** [comment]
@gargcreation1992 This pull request only adds alias, nothing more. Logic is covered by tests, parameters are checked in the `limelightDigital` adapter. Should it have tests for alias params? I have not met such a requirement before

**Comment by gargcreation1992** [comment]
> @gargcreation1992 This pull request only adds alias, nothing more. Logic is covered by tests, parameters are checked in the `limelightDigital` adapter. Should it have tests for alias params? I have not met such a requirement before

agreed. I was insisting since you have already added the test cases for bidder params. Since that is covered by the `limelightDigital` adapter, you do not need to add it for an alias. Approving the PR now.

**Review by gargcreation1992 — APPROVED** [review]

**Review by onkarvhanumante — APPROVED** [review]

**Pull request merged**
